packages/vscode/src/plugins/prisma-language-server/index.ts

```diff
@@ -73,21 +89,39 @@ const startFileWatcher = (rootPath: string) => {
     depth: 9,
     recursive: true,
     ignoreInitial: true,
-    ignore: (targetPath) => {
-      if (targetPath === rootPath) {
-        return false
+    ignore(targetPath) {
+      if (!targetPath.startsWith(rootPath)) return true
+      if (targetPath.endsWith('.git')) return true
+
+      // every time we hit a project path, ensure some sub-paths are whitelisted
+      if (!(targetPath in allowedWatcherPaths) && fs.statSync(targetPath).isDirectory()) {
+        try {
+          // we get the location of installation of the @prisma/client package
+          const clientPath = path.dirname(require.resolve('@prisma/client', { paths: [targetPath] }))
```

Comment:
This is the trick not to watch the whole node_modules.

Reply:
Isn't this super dangerous from a performance perspective as it will be run for _every_ file change, all the time?

Reply:
Hm, one thing I'm immediately noting is that the initial run through scans a lot more paths

https://github.com/prisma/language-tools/assets/29753584/352abe77-e727-422d-8e48-728272ef37b6

Reply:
I'm really confused about it including this too (one level above the repo)
```
targetpath : /Users/soph/src
targetpath : /Users/soph/src
```

Reply:
🤔 
```ts
if (targetPath.endsWith('.git')) return true
```

<img width="458" alt="image" src="https://github.com/prisma/language-tools/assets/29753584/c6bf7a75-7605-4f59-9c19-f0aba574c80a">